flytekit/types/file/file.py

```diff
@@ -732,25 +735,28 @@ async def async_to_python_value(
 
         # For the remote case, return an FlyteFile object that can download
         local_path = ctx.file_access.get_random_local_path(uri)
+
         expected_format = FlyteFilePathTransformer.get_format(expected_python_type)
         ff = FlyteFile.__class_getitem__(expected_format)(
-            path=local_path, downloader=lambda: self.downloader(ctx=ctx, remote_path=uri, local_path=local_path)
+            path=local_path,
+            downloader=partial(self.downloader, ctx.file_access, remote_path=uri, local_path=local_path),
         )
         ff._remote_source = uri
-
         return ff
 
     @staticmethod
     def downloader(
-        ctx: FlyteContext, remote_path: typing.Union[str, os.PathLike], local_path: typing.Union[str, os.PathLike]
+        file_access_provider: FileAccessProvider,
```

Comment:
This breaks backward compatibility. Is it safe to change the signature here?

Reply:
As far as I can tell this staticmethod was added in a PR 4 days ago: https://github.com/flyteorg/flytekit/pull/2991/files

And the only call site was in this module itself

Reply:
we can also revert this to the older implementation with the pure partial

Reply:
That works for me.